src/ophyd_async/core/_providers.py

```diff
@@ -213,5 +213,9 @@ def __call__(self) -> str:
 
 class ShapeProvider(Protocol):
     @abstractmethod
-    async def __call__(self) -> tuple:
+    async def np_datatype(self) -> str:
+        """Represents the numpy datatype"""
+
+    @abstractmethod
+    async def __call__(self) -> tuple[int, ...]:
```

Comment:
I'd be tempted to make this more descriptive now we have 2 things we want out of ShapeProvider
```suggestion
    async def shape(self) -> tuple[int, ...]:
```

Reply:
At which point, really is a ShapeProvider not a DataDescriber or similar?

Reply:
I made a prospective PR into this PR in case we want to make that change, but I want to get something merged today so's to enable dodal update ASAP. https://github.com/bluesky/ophyd-async/pull/556